ospfd/ospf_vty.c

```diff
@@ -10724,18 +10725,21 @@ static void show_ip_ospf_route_network(struct vty *vty, struct ospf *ospf,
 			}
 			break;
 		case OSPF_PATH_INTRA_AREA:
+			or_is_transit = (or->type == OSPF_DESTINATION_NETWORK);
 			if (json) {
 				json_object_string_add(json_route, "routeType",
-						       "N");
+						      or_is_transit && detail ?
+							                "N T" :
+								        "N");
 				json_object_int_add(json_route, "cost",
 						    or->cost);
 				json_object_string_addf(json_route, "area",
 							"%pI4",
 							&or->u.std.area_id);
 			} else {
-				vty_out(vty, "N    %-18s    [%d] area: %pI4\n",
-					buf1, or->cost,
-					&or->u.std.area_id);
+				vty_out(vty, "N %s  %-18s    [%d] area: %pI4\n",
+					or_is_transit && detail ? "T" : " ",
```

Comment:
Maybe we can give a more descriptive information (T - transitive, N - ...) into a header or so?

Reply:
> Maybe we can give a more descriptive information (T - transitive, N - ...) into a header or so?

like this?

```
bsp-debianrt-exp1# show ip ospf route detail
============ OSPF network routing table ============
N T  10.0.0.0/24           [32] area: 0.0.0.0
                           via 192.168.124.67, ens192
                           adv 10.0.0.5
N    10.0.30.0/24          [33] area: 0.0.0.0
                           via 192.168.124.67, ens192
                           adv 10.0.0.5
...
============ OSPF external routing table ===========
N - network         T - transitive  IA - intra-area
E - external route  D - destination R  - router
```

Reply:
Yes, but at the top of the output :)

Reply:
> Yes, but at the top of the output :)

Ok, I change the output format it into this.

```
bsp-debianrt-exp1# show ip ospf rou de
N - network         T - transitive  IA - intra-area
E - external route  D - destination R  - router

============ OSPF network routing table ============
N T  10.0.0.0/24           [32] area: 0.0.0.0
                           via 192.168.124.67, ens192
                           adv 10.0.0.5
N T  10.0.30.0/24          [33] area: 0.0.0.0
                           via 192.168.124.67, ens192
                           adv 10.0.0.5
```